kasa/experimental/sslaestransport.py

```diff
@@ -406,19 +406,41 @@ async def perform_handshake1(self) -> tuple[str, str, str]:
         _LOGGER.debug(msg)
         raise AuthenticationError(msg)
 
-    def _handshake_session_expired(self):
-        """Return true if session has expired."""
-        return (
-            self._session_expire_at is None
-            or self._session_expire_at - time.time() <= 0
+    async def try_send_handshake1(self, username: str, local_nonce: str) -> dict:
+        """Perform the handshake."""
+        _LOGGER.debug("Will to send handshake1...")
+
+        # Device needs the content length or it will response with 500
+        body = {
+            "method": "login",
+            "params": {
+                "cnonce": local_nonce,
+                "encrypt_type": "3",
+                "username": self._username,
+            },
+        }
+        http_client = self._http_client
+
+        status_code, resp_dict = await http_client.post(
+            self._app_url,
+            json=body,
+            headers=self._headers,
+            ssl=await self._get_ssl_context(),
         )
 
+        _LOGGER.debug("Device responded with: %s", resp_dict)
+
+        if status_code != 200:
+            raise KasaException(
+                f"{self._host} responded with an unexpected "
+                + f"status code {status_code} to handshake1"
+            )
+
+        return cast(dict, resp_dict)
+
     async def send(self, request: str) -> dict[str, Any]:
         """Send the request."""
-        if (
-            self._state is TransportState.HANDSHAKE_REQUIRED
-            or self._handshake_session_expired()
```

Comment:
These sessions do not expire?

Reply:
They might do as there is a `SESSION_EXPIRED` error code but there's nothing to tell us when it expires so we'll just handle that error code.